### PR TITLE
Port get_link_name () and move to FileUtils

### PR DESCRIPTION
**Pull request by jeremypw**
The function `get_link_name ()` is ported to Vala from `marlin_file_operations.c` and moved to FileUtils.vala.

The format for multiple links has been simplified as discussed.

The associated function `shorten_utf8_string ()` was also ported.

Unit tests included.

### CODE REVIEW

File: libcore/FileUtils.vala [hunk 1]
```diff
@@ -891,6 +891,81 @@ namespace PF.FileUtils {
...
+                    ///TRANSLATORS: %'d represents 22 or 32 or 42 etc
+                    result = _("%'dnd link to %s").printf (count, target_name);
+                    break;
+                case 3:
+                    ///TRANSLATORS: %'d represents 23 or 33 or 43 etc
+                    result = _("%'drd link to %s").printf (count, target_name);
+                    break;
+                default:
+                    ///TRANSLATORS: %'d represents a number in range 4 - 9, 14 - 19, 24 - 29 etc
+                    result = _("%'dth link to %s").printf (count, target_name);
```
Comment (danrabbit) [inline libcore/FileUtils.vala:958]: This doesn't seem right at all for localization. I'm actually not sure of the right way to do this though. Maybe @ryonakano knows

**Review by danrabbit — DISMISSED** [review]
While we're here, I think we should see if this is really the right way to localize these strings

File: libcore/FileUtils.vala (hunk 1 above)
Comment (danrabbit) [reply libcore/FileUtils.vala:958]: @jeremypw Maybe we can vastly simplify this and go for "FileName (link)", "FileName (link 2)", etc. I think I saw somewhere you were looking at the copy one too. We could apply the same pattern and do "FileName (copy)", "FileName (copy 2)", etc.

Comment (jeremypw) [reply libcore/FileUtils.vala:958]: Yes, I am happy to do that - I was thinking there was a lot of fairly involved code for not very much gain.

File: libcore/FileUtils.vala [hunk 2]
```diff
@@ -919,6 +919,52 @@ namespace PF.FileUtils {
...
+            case 0:
+                result = target_name; //First link in directory has same name as source
+                break;
+
+            case 1:
+                result = _("%s (link)").printf (target_name);
+                break;
+
+            default:
+                result = _("%s (link %i)").printf (target_name, count);
```
Comment (tintou) [inline libcore/FileUtils.vala:957]:
```suggestion
                result = _("%s (link %d)").printf (target_name, count);
```
We're using %d everywhere ;)

**Comment by tintou** [comment]
Ready to go on my side, neat that you added some tests!

File: libcore/FileUtils.vala (hunk 1 above)
Comment (jeremypw) [reply libcore/FileUtils.vala]: Naming now simpler

**Review by tintou — APPROVED** [review]

**Pull request merged**